amazon_features/steps/amazon.py

```diff
@@ -0,0 +1,45 @@
+from behave import given, when, then
+
+from features.steps.functions import send_input, click, select_items_extract_prices, total_cart_prices
+
+"""......................all the locators......................"""
+
+tags = {"SEARCH BOX": "xpath=//*[@id='twotabsearchtextbox']",
+        "SEARCH": "xpath=//*[@id='nav-search-submit-button']",
+        "RATING": "xpath =(//ul[@data-csa-c-content-id='1318475031']/span)",
+        "ITEMS LIST": "xpath=//span[@class='a-size-medium a-color-base a-text-normal']",
+        "ADD TO CART BTN": "xpath=//*[@id='add-to-cart-button']",
+        "CART ICON": "xpath=//*[@class='nav-cart-icon nav-sprite']",
+        "TOTAL CART PRICE": "xpath=//*[@id='sc-subtotal-amount-buybox']/span",
+        "ITEM PRICE": "xpath=//span[@id='tp_price_block_total_price_ww']/descendant::span[@class='a-price-whole']"
+        }
+""".............................................................."""
+
+
+@given(u'User is on the amazon website search for "{search_query}"')
+def step_impl(context, search_query):
+    send_input(context.page, tags["SEARCH BOX"], search_query)  # user entered search item and searched for product
+    click(context.page, tags["SEARCH"])
+
+
+@when(u'User filter by "{star}" ratings')
+def filtering_the_items(context, star):
+    Dict = {"4": 1, "3": 2, "2": 3, "1": 4}  # created a dict to select the locator according to the user's input
```

Comment:
What is the purpose of the dictionary here ?,
can't we use the rating number as XPath ?